src/simple-todos/step07/imports/ui/App.svelte

```diff
@@ -22,22 +21,19 @@
     const pendingOnlyFilter = { ...hideCompletedFilter, ...userFilter };
 
 
-    tasks = user
+    getTasks = user
```

Comment:
I'm wondering if it is better to use the sync Mongo api in Svelte components. I am planning to do that in my projects. The async api adds a lot of complexity here, but doesn't seem to have any benefits. Meteor will not be able to remove the sync Mongo api from the client for a long time if ever since there are too many places it is used that only support sync code.

Reply:
Yeah. I thought a bit about this, but I also liked using promises with Svelte. Guess I could show that it can be used both ways. Using promises feels like when we are creating UI in other places (Svelte /SvelteKit).
Don't know what to do when the release is launched. What is your opinion @zodern?

Reply:
The async api seems unnecessary since the client isn't waiting for anything. It works the same as the sync api, but you now have to wait for the promise to resolve.

It isn't uncommon to run a query, and then use the results within the same or another reactive/tracker statement. With the async api it isn't easy to do. This is a larger problem because you can't use Tracker.autorun in all of the places svelte's normal dependency tracking works, so how you would normally handle this in svelte might not work with Meteor. I'm looking into how we can allow using Tracker.autorun with `@const` tags, which will partially help with this.

Another minor thing is using `#await` encourages creating a loading state, but it will never be shown since the promise should resolve before Svelte updates the DOM. If I am wrong and the loading state is briefly shown each time the promise is re-created then that could be a performance and UX problem.

Reply:
I looked into it a little more, and it seems every time the promise is re-created, svelte removes all the DOM from the {:then} block, creates the DOM for the loading state, and then in the next micro task removes the DOM for the loading state and recreates the {:then} block.

I created an example in the svelte REPL to show this: https://svelte.dev/repl/c398061483f247b9a77bf767c6d3ab40?version=3.50.1
To make it easier to see what is happening, I added a fade transition to the loading and {:then} blocks.

Without a transition, I'm not able to find a way for the loading state to ever be visible. The browser should run all micro tasks before updating the UI, so it should be impossible for it to ever be visible.

Reply:
I've seen it now. Not sure what to do. Perhaps In the tutorial use?
```sveltehtml
{#await promise then value}
	<p>the value is {value}</p>
{/await}
```
At some point, those Database methods will be async (if I understood correctly [here](https://github.com/meteor/meteor/pull/12028)).
what would you say is the best way to handle this matter then?

Reply:
That would be a less verbose way of doing it. Another option is to do something like:
```
let tasks = [];
$m: {
  Tasks.find().fetchAsync().then(value => tasks = value);
}
```

This would be significantly more efficient compared to `#await` when the data frequently changes, and allow you to use the data in other reactive or tracker statements.

> At some point, those Database methods will be async (if I understood correctly https://github.com/meteor/meteor/pull/12028).
what would you say is the best way to handle this matter then?

The sync api will stop working on the server when Fibers are unavailable, but the plan is to keep it on the client, possibly forever, since there are too many places that can only use the sync api including Blaze helpers and Tracker.autorun.